beancount_reds_importers/libreader/csv_multitable_reader.py

```diff
@@ -69,18 +67,25 @@ def read_file(self, file):
 
         self.raw_rdr = rdr = self.read_raw(file)
 
+        # skip non-table rows at file head & footer
+        skip_head_rows = getattr(self, 'skip_head_rows', 0)
+        skip_tail_rows = getattr(self, 'skip_tail_rows', 0)
+        rdr = rdr.skip(skip_head_rows) # chop unwanted header rows
+        rdr = rdr.head(len(rdr) - skip_tail_rows - 1)  # chop unwanted footer rows
+
         #     [0, 2, 10] <-- starts
         # [-1, 1, 9]     <-- ends
         table_starts = [i for (i, row) in enumerate(rdr) if self.is_section_title(row)] + [len(rdr)]
         table_ends = [r-1 for r in table_starts][1:]
         table_indexes = zip(table_starts, table_ends)
+        skip_section_rows = getattr(self, 'skip_section_rows', 1)
 
         # build the dictionary of tables
         self.alltables = {}
         for (s, e) in table_indexes:
             if s == e:
                 continue
-            table = rdr.skip(s+1)
+            table = rdr.skip(s+skip_section_rows)
```

Comment:
Hmm, I'm not sure I understand what skip_section_rows is?

Reply:
`skip_section_rows` is the number of lines to initially skip after a starting index of a table. It's naming was intended to go in hand with `is_section_title`, but could rename to `skip_table_head_rows` if that feels more consistent.

For schwab balances parsing it's presumably 1, for how I was initially doing fidelity csv parsing with csv_multitable it was 0. So it would look something like below:
_(An aside: For the [fidelity_csv_old](https://github.com/savingsandloan/beancount_reds_importers/commits/fidelity_csv_old) importer I used `csv_multitable` at the time as it seemed like a good interface to layer in other tables of information for generated account open/close & commodity listing information)_

```
Run Date,Action,Symbol,Security Description,Security Type,Quantity,Price ($),Commission ($),Fees ($),Accrued Interest ($),Amount ($),Settlement Date
 12/01/2021, TRANSFERRED TO VS 222-222222-2 CUR YR PARTICPNT CON (Cash), , No Description,Cash,,,,,,-888,
 11/22/2021, DIRECT DEPOSIT HEALTHHELLHealthHell (Cash), , No Description,Cash,,,,,,1100.00,
```
...and the `skip_section_rows` if you're detecting `is_section_title` based on the line `Run Date,Action,Symbol,Security Description,Security Type,Quantity,Price ($),Commission ($),Fees ($),Accrued Interest ($),Amount ($),Settlement Date`. If it was hardcoded to 1 you'd end up skipping over the first row of data (in this example `12/01/2021, TRANSFERRED...`)

Supposedly one could avoid needing this by improving logic around finding `table_starts` indexes to do something more sophisticated than `is_section_title` (which just evaluates a single row), at the time it seems easier to just have a configurable variable.

Reply:
I can see that. But given the need for skip_section_rows doesn't seem common, I'd rather keep `csv_multitable_reader.py` simple, and leave it out of it, and instead treat it as an institution-specific quirk. This way, the method can be overridden for institutions where it appears while keeping common code complexity down, if that makes sense.

Reply:
Ok, removed it for now, on my end will need to do something different than using `csv_multitable_reader` for tracking the different tables of info.

Reply:
Do you have an example multitable csv where this is an issue? fidelity_csv should use the single table reader, right?

Reply:
Yeah, the fidelity CSV is a single table (thus generally makes sense for a single table reader), though in seeking feature additions around generating account/commodity entries (as separate tables) I was starting to creatively use the multitable csv reader because of it's output structure as a table and calling of `prepare_tables`. 

To get a glimpse in `fidelity_csv_old` I was using csv_multitable_reader to leverage the `prepare_tables` call to generate other tables (accounts, commodity entries) based on the transactions table: https://github.com/savingsandloan/beancount_reds_importers/blob/8e884084df35ceaa89f3c350eb454efeb37297b6/beancount_reds_importers/fidelity_csv/__init__.py#L898

Again, there's probably a different way for me to do this without `csv_multitable_reader` if its intent is to follow a specific format around section titles & headings, but it helped at the time with framing the structure of data in the custom importer.

Reply:
Got it. Yes, I'd definitely recommend fully separating those concerns (file format reader vs. data structures).